web/components/App.jsx

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable no-console */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import _ from 'lodash';
+import FilterHeader from './FilterHeader.jsx';
+import EventMessage from './EventMessage.jsx';
+import UserMessage from './UserMessage.jsx';
+import ChatMessage from './ChatMessage.jsx';
+import AttachmentMessage from './AttachmentMessage.jsx';
+import ToastMessage from './ToastMessage.jsx';
+import ChatBox from './ChatBox.jsx';
+import './chat.css';
+
+const { env } = require('../../config.js');
+const config = require('../../config.js')[env];
+const botName = require('../../package.json').name;
+const bdk = require('@salesforce/refocus-bdk')(config, botName);
+import MessageService from '../services/MessageService';
+import AttachmentService from '../services/AttachmentService';
+
+const ZERO = 0;
+const DFB = 5; // Distance From Bottom
+const messageService = new MessageService(bdk);
+const attachmentService = new AttachmentService(bdk);
+
+/**
+ * @param {object} props
+ * @returns {JSX} app container.
+ */
+export default function App(props) {
+  let container;
+  const { response } = props;
+  const [toast, setToast] = useState(false);
+  const [events, setEvents] = useState(response);
+  const [currentText, setCurrentText] = useState('');
+  const [scroll, setScroll] = useState(true);
+  const [fileDraggedOver, setFileDraggedOver] = useState(false);
+  const [filter, setFilter] = useState('CommentAttachment');
+  const [pendingMessage, setPendingMessage] = useState(false);
+
+  /**
+   * @param {array} previousEvents
+   * @param {array} newEvents
+   */
+  function handleNewEventsReceived(previousEvents, newEvents) {
+    const allEvents = previousEvents.concat(newEvents).filter((value, index, self) => {
+      const duplicates = _.filter(self.slice(ZERO, index), ['id', value.id]);
```

Comment:
Do you need lodash to filter? Not familiar with lodash, but there's a lot out of the box with es6 no?